File: server/bcephal-base/src/main/java/com/moriset/bcephal/domain/socket/TaskProgressInfo.java
/**
 * 
 */
package com.moriset.bcephal.domain.socket;

import lombok.Data;

/**
 * @author Moriset
 *
 */
@Data
public class TaskProgressInfo {

	private Long id;

	private String name;

	private boolean taskEnded;

	private boolean taskInError;

	private String message;

	private long stepCount;

	private long currentStep;

	private TaskProgressInfo currentSubTask;	
	
}

File: client/bcephal-client-model/Base/TaskProgressInfo.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bcephal.Models.Base
{
    public class TaskProgressInfo
    {

		public long? Id { get; set; }

		public string Name { get; set; }

		public bool TaskEnded { get; set; }

		public bool TaskInError { get; set; }

		public string Message { get; set; }

		public long StepCount { get; set; }

		public long CurrentStep { get; set; }

		public TaskProgressInfo CurrentSubTask { get; set; }

		[JsonIgnore]
		public string ProgressLabel
        {
            get
            {
				long rate = StepCount != 0 ? (Int32)(CurrentStep * 100 / StepCount) : 0;
				if(rate > 100)
                {
					rate = 100;
                }
				return "" + rate + " % (" + CurrentStep + "/" + StepCount + ")";

			}
        }

	}
}
